pkg/logs/auditor/auditor.go

```diff
@@ -161,13 +163,15 @@ func (a *RegistryAuditor) run() {
 	for {
 		select {
 		case <-a.health.C:
-		case msg, isOpen := <-a.inputChan:
+		case payload, isOpen := <-a.inputChan:
 			if !isOpen {
 				// inputChan has been closed, no need to update the registry anymore
 				return
 			}
 			// update the registry with new entry
-			a.updateRegistry(msg.Origin.Identifier, msg.Origin.Offset, msg.Origin.LogSource.Config.TailingMode, msg.IngestionTimestamp)
+			for _, msg := range payload.Messages {
+				a.updateRegistry(msg.Origin.Identifier, msg.Origin.Offset, msg.Origin.LogSource.Config.TailingMode, msg.IngestionTimestamp)
```

Comment:
More of an inconsequential architectural question but could this benefit from a fan-out goroutine for the `updateRegistry` call? I'm guessing there's no benefit to doing so but just wanted to check.

Reply:
probably not. This is very light weight and the updates are all done in-memory.